### PR TITLE
Cmake build system 2

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

**Review by vlkale — COMMENTED** [review]
Waiting for other's review and then this should be ready for merging into Kokkos Develop

**Comment by crtrott** [comment]
Ok I think this is good enough to merge for now.

File: Build.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+1. Create your build directory and go to it
+
+2. Type `ccmake ..`  and change any options, including tools you want turned on  (some are by default off).  (Optional)
+
+3. Type `cmake ..` 
+
+4. Type `make` 
+
+5. Specify the generated .dylib file in the environment variable KOKKOS_TOOLS_LIBRARY when running your Kokkos-based application.
```
Comment (masterleinad) [inline Build.md:13]: You only get `*dylib` files for `Mac`, right?

```suggestion
5. Specify the generated shared library in the environment variable KOKKOS_TOOLS_LIBRARY when running your Kokkos-based application. 
```

File: Build.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+5. Specify the generated .dylib file in the environment variable KOKKOS_TOOLS_LIBRARY when running your Kokkos-based application. 
+
+
+# With Makefile (recommended)
+
+1. Go into the directory of the particular tool, e.g., `cd debugging/kernel_logger`
+
+2. Type `make` 
+
+3. Specify the generated .so file in the environment variable KOKKOS_TOOLS_LIBRARY when running your Kokkos-based application.
```
Comment (masterleinad) [inline Build.md:22]: Step 3 should be the same for both setups.

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+# Disable in-source builds to prevent source tree corruption.
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
+  message(FATAL_ERROR "FATAL: In-source builds are not allowed. You should create a separate directory for build files.")
+endif()
+
+list(INSERT CMAKE_MODULE_PATH 0 ${PROJECT_SOURCE_DIR}/cmake)
+
+message(STATUS)
+message(STATUS Configuring Kokkos-Tools)
+message(STATUS)
```
Comment (masterleinad) [inline CMakeLists.txt:29]: I t looks weid to have a `message(STATUS)` without any message to print.
```suggestion
message(STATUS "Configuring Kokkos-Tools\n")
```

File: cmake/FindITT.cmake [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# Note: Package is named "ITT" here because we reuse Caliper's FindITTAPI.cmake find module
+# and it calls find_package_handle_standard_args() with "ITT" package name internally, so CMake
+# expectes find_package() calles to use "ITT" package name as well.
```
Comment (masterleinad) [inline cmake/FindITT.cmake:3]:
```suggestion
# expectes find_package() calls to use "ITT" package name as well.
```

**Comment by vlkale** [comment]
@crtrott  OK, I see the changes, including the fixes to the Makefiles made. 


Waiting for another review (from @dalg24) in order to merge.

**Review by dalg24 — APPROVED** [review]

**Comment by dalg24** [comment]
Acknowledging I saw Daniels comment but let us just fix that as a follow-up.

**Pull request merged**